vendor/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -68,32 +71,75 @@ var _ recognizer.RecognizingDecoder = &Serializer{}
 
 func init() {
 	// Force jsoniter to decode number to interface{} via ints, if possible.
-	// decodeNumberAsInt64IfPossible := func(ptr unsafe.Pointer, iter *jsoniter.Iterator) {
-	// 	switch iter.WhatIsNext() {
-	// 	case jsoniter.NumberValue:
-	// 		var number json.Number
-	// 		iter.ReadVal(&number)
-	// 		u64, err := strconv.ParseUint(string(number), 10, 64)
-	// 		if err == nil {
-	// 			*(*interface{})(ptr) = u64
-	// 			return
-	// 		}
-	// 		i64, err := strconv.ParseInt(string(number), 10, 64)
-	// 		if err == nil {
-	// 			*(*interface{})(ptr) = i64
-	// 			return
-	// 		}
-	// 		f64, err := strconv.ParseFloat(string(number), 64)
-	// 		if err == nil {
-	// 			*(*interface{})(ptr) = f64
-	// 			return
-	// 		}
-	// 		// Not much we can do here.
-	// 	default:
-	// 		*(*interface{})(ptr) = iter.Read()
-	// 	}
-	// }
-	// jsoniter.RegisterTypeDecoderFunc("interface {}", decodeNumberAsInt64IfPossible)
+	decodeNumberAsInt64IfPossible := func(ptr unsafe.Pointer, iter *jsoniter.Iterator) {
+		switch iter.WhatIsNext() {
+		case jsoniter.NumberValue:
+			var number json.Number
+			iter.ReadVal(&number)
+			i64, err := strconv.ParseInt(string(number), 10, 64)
+			if err == nil {
+				*(*interface{})(ptr) = i64
+				return
+			}
+			f64, err := strconv.ParseFloat(string(number), 64)
+			if err == nil {
+				*(*interface{})(ptr) = f64
+				return
+			}
```

Comment:
compare upstream (1.13 PR) https://github.com/kubernetes/kubernetes/pull/83436/commits/a4e734ac9fea9affdffee9ac845c15e1abacfab3

Upstream has a DecodeNumber error here.

Reply:
I tried to put in minimal change that would meet the intent of the upstream change, i.e. not perturb the original code if it wasn't involved in the new logic. I thought the core logic for upstream change is by adding 'default:' case and checking increment depth. Why mess with case NumberValue?